core/storage/base_model/gae_models.py

```diff
@@ -687,17 +711,15 @@ def has_reference_to_user_id(cls, user_id: str) -> bool:
         """
         return cls.query(cls.user_id == user_id).get(keys_only=True) is not None
 
-    # TODO(#13523): Change 'commit_cmds' to domain object/TypedDict to
-    # remove Any from type-annotation below.
     @classmethod
     def create(
             cls: Type[SELF_BASE_COMMIT_LOG_ENTRY_MODEL],
             entity_id: str,
             version: int,
             committer_id: str,
             commit_type: str,
-            commit_message: str,
-            commit_cmds: Union[Dict[str, Any], List[Dict[str, Any]], None],
+            commit_message: Optional[str],
+            commit_cmds: BaseCommitLogEntryCmdType,
```

Comment:
This reads oddly to me because the type seems singular but commit_cmds should be a list of items.

Ditto elsewhere.

Reply:
The name is Changed to match the commit command's type.
Also, I used a type variable here because the same type is repeated in all files.

Reply:
Something's wrong with the typing for BaseLogEntryAllowedCommitCmdTypes in core/storage/base_model/gae_models.py -- you basically say it's a union of a list(dict) and a dict. It can't represent both, it should be either singular or plural.

It probably makes sense to have just one Type for the mapping and, in places where you are working with a list of those, type that as Sequence[Type]. I don't think you need VersionedAllowedCommitCmdTypes either since that's basically Sequence[Type].

Reply:
In some base_test cases, `commit_cmds` is initialized as `empty dict` due to this I have to include mapping as an individual type in BaseLogEntryAllowedCommitCmdTypes but after looking at docstring and other model's implementations I think in test cases it should be List[Dict[...]] not only Dict[].

So, now dict is removed from BaseLogEntryAllowedCommitCmdTypes. Now we know List is used throughout the codebase so I define a single type variable and used it everywhere.